.github/workflows/pr-merged.yml

```diff
@@ -0,0 +1,192 @@
+          git diff-index --quiet HEAD ./quickstarts/* || git commit -m 'chore: generate UUID(s) [skip ci]'
+          echo "::set-output name=commit::true"
+
+      - name: Temporarily disable branch protections
+        id: disable-branch-protection
+        if: always()
+        uses: actions/github-script@v1
+        with:
+          github-token: ${{ secrets.OPENSOURCE_BOT_TOKEN }}
+          previews: luke-cage-preview
+          script: |
+            const result = await github.repos.updateBranchProtection({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              branch: 'main',
+              required_status_checks: null,
+              restrictions: null,
+              enforce_admins: null,
+              required_pull_request_reviews: null
+            })
+            console.log("Result:", result)
+
+      - name: Push Commit
+        if: steps.commit-changes.outputs.commit == 'true'
+        run: git push origin HEAD
+
+      - name: Re-enable branch protections
+        id: enable-branch-protection
+        if: always()
+        uses: actions/github-script@v1
+        with:
+          github-token: ${{ secrets.OPENSOURCE_BOT_TOKEN }}
+          previews: luke-cage-preview
+          script: |
+            const result = await github.repos.updateBranchProtection({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              branch: 'main',
+              required_status_checks: {
+                strict: true,
+                contexts: []
+              },
+              restrictions: {
+                "users":[],
+                "teams":[],
+                "apps":[]
+              },
+              enforce_admins: null,
+              required_pull_request_reviews: {
+                dismiss_stale_reviews: true,
+                required_approving_review_count: 1,
+                dismissal_restrictions: {
+                  users: [],
+                  teams: []
+                }
+              }
+            })
+            console.log("Result:", result)
+
+  submit-install-plans:
+    name: Submit install plans
+    needs: [generate-uuid]
+    if: ${{github.event.pull_request.merged}}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        # Ensure we have the most recent commit to `main`
+        with:
+          ref: "main"
+          fetch-depth: 0
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+
+      - name: Install dependencies
+        run: cd utils && yarn install
+      - name: Update install plans - staging
+        env:
+          NR_API_URL: ${{ secrets.NR_API_URL_STAGING }}
+          NR_API_TOKEN: ${{ secrets.NR_API_TOKEN_STAGING }}
+        run: |
+          URL="https://api.github.com/repos/${GITHUB_REPOSITORY}/pulls/${{ github.event.pull_request.number }}/files"
+          DRY_RUN=false
```

Comment:
If we wanted to reduce some of the duplication, we might be able to set these variables at the workflow level (rather than at the step level).

Reply:
so true!